cmd/dataprotection/main.go

```diff
@@ -0,0 +1,359 @@
+		// when the Manager ends. This requires the binary to immediately end when the
+		// Manager is stopped, otherwise, this setting is unsafe. Setting this significantly
+		// speeds up voluntary leader transitions as the new leader doesn't have to wait
+		// LeaseDuration time first.
+		//
+		// In the default scaffold provided, the program ends immediately after
+		// the manager stops, so would be fine to enable this option. However,
+		// if you are doing or intending to do any operation such as performing cleanups
+		// after the manager stops then its usage might be unsafe.
+		LeaderElectionReleaseOnCancel: true,
+
+		CertDir:               viper.GetString("cert_dir"),
+		ClientDisableCacheFor: intctrlutil.GetUncachedObjects(),
+	})
+	if err != nil {
+		setupLog.Error(err, "unable to start manager")
+		os.Exit(1)
+	}
+
+	if viper.GetBool(dataProtectionFlagKey.viperName()) {
+		if err = (&dpcontrollers.BackupToolReconciler{
+			Client:   mgr.GetClient(),
+			Scheme:   mgr.GetScheme(),
+			Recorder: mgr.GetEventRecorderFor("backup-tool-controller"),
+		}).SetupWithManager(mgr); err != nil {
+			setupLog.Error(err, "unable to create controller", "controller", "BackupTool")
+			os.Exit(1)
+		}
+
+		if err = (&dpcontrollers.BackupPolicyReconciler{
+			Client:   mgr.GetClient(),
+			Scheme:   mgr.GetScheme(),
+			Recorder: mgr.GetEventRecorderFor("backup-policy-controller"),
+		}).SetupWithManager(mgr); err != nil {
+			setupLog.Error(err, "unable to create controller", "controller", "BackupPolicy")
+			os.Exit(1)
+		}
+
+		if err = (&dpcontrollers.CronJobReconciler{
+			Client:   mgr.GetClient(),
+			Scheme:   mgr.GetScheme(),
+			Recorder: mgr.GetEventRecorderFor("cronjob-controller"),
+		}).SetupWithManager(mgr); err != nil {
+			setupLog.Error(err, "unable to create controller", "controller", "CronJob")
+			os.Exit(1)
+		}
+
+		if err = (&dpcontrollers.BackupReconciler{
+			Client:   mgr.GetClient(),
+			Scheme:   mgr.GetScheme(),
+			Recorder: mgr.GetEventRecorderFor("backup-controller"),
+		}).SetupWithManager(mgr); err != nil {
+			setupLog.Error(err, "unable to create controller", "controller", "Backup")
+			os.Exit(1)
+		}
+
+		if err = (&dpcontrollers.RestoreJobReconciler{
+			Client:   mgr.GetClient(),
+			Scheme:   mgr.GetScheme(),
+			Recorder: mgr.GetEventRecorderFor("restore-job-controller"),
+		}).SetupWithManager(mgr); err != nil {
+			setupLog.Error(err, "unable to create controller", "controller", "RestoreJob")
+			os.Exit(1)
+		}
+
+		if err = (&dpcontrollers.BackupRepoReconciler{
+			Client:   mgr.GetClient(),
+			Scheme:   mgr.GetScheme(),
+			Recorder: mgr.GetEventRecorderFor("backup-repo-controller"),
+		}).SetupWithManager(mgr); err != nil {
+			setupLog.Error(err, "unable to create controller", "controller", "BackupRepo")
+			os.Exit(1)
+		}
+	}
+
+	if viper.GetBool(storageFlagKey.viperName()) {
+		if err = (&storagecontrollers.StorageProviderReconciler{
```

Comment:
why launch a storage provider controller?

Reply:
Remove the storage provider controller from dataProtection, that will be launched by manager.